AdminServer/appscale/admin/push_worker_manager.py

```diff
@@ -101,6 +107,26 @@ def update_worker(self, queue_config):
     yield gen.with_timeout(timedelta(seconds=60), start_future,
                            IOLoop.current())
 
+    try:
+      yield self.ensure_pid_changed(old_pid, pid_location)
+    except AssertionError:
+      # Occasionally, Monit will get interrupted during a restart. Retry the
+      # restart if the Celery worker PID is the same.
+      logger.warning(
+        '{} worker PID did not change. Restarting it.'.format(self.project_id))
+      yield self.update_worker(queue_config)
+
+  @staticmethod
+  @retry_coroutine(retrying_timeout=10, retry_on_exception=[AssertionError])
+  def ensure_pid_changed(old_pid, pid_location):
+    try:
+      with open(pid_location) as pidfile:
+        new_pid = int(pidfile.read().strip())
+    except IOError:
+      new_pid = None
+
+    assert new_pid != old_pid
```

Comment:
Though I like implementation with assert statement, Codacy raised a good point, that assert statements can be ignored in some interpreters for optimization purposes.